generated_code/aderdg.py

```diff
@@ -197,11 +197,15 @@ def starMatrix(self, dim):
     pass
 
   def addInit(self, generator):
+    godunov_spp = self.godunov_spp()
+    self.QcorrLocal = Tensor('QcorrLocal', godunov_spp.shape)
+    self.QcorrNeighbor = Tensor('QcorrNeighbor', godunov_spp.shape)
+
     fluxScale = Scalar('fluxScale')
-    computeFluxSolverLocal = self.AplusT['ij'] <= fluxScale * self.Tinv['ki'] * self.QgodLocal['kq'] * self.starMatrix(0)['ql'] * self.T['jl']
+    computeFluxSolverLocal = self.AplusT['ij'] <= fluxScale * self.Tinv['ki'] * (self.QgodLocal['kq'] * self.starMatrix(0)['ql'] + self.QcorrLocal['kl']) * self.T['jl']
```

Comment:
how do adding these zeros in the case of Godunov ( extra flops) will affect performance?

Reply:
That's only done once during the initialization procedure. The actual computation is not modified; we effectively "just" exchange some matrices.